### PR TITLE
Codechange: Cleaned up and renamed NodeList

### PR DESCRIPTION
**Pull request by Kuhnovic**
## Motivation / Problem

The class CNodeList_HashTableT doesn't follow our coding style and could use a little bit of TLC.

## Description

Renamed CNodeList_HashTableT to NodeLIst, cleaned up some functions and did a bit of renaming. Changed some of the comments to better describe what it actually does.

Note that I specifically focused on improving nodelist.hpp, all other changes are collateral damage. Wherever I changed other files I tried to make things (more) coding-style-compliant without overhauling the entire file.

## Limitations

None

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/pathfinder/aystar.cpp [hunk 1]
```diff
@@ -25,11 +25,11 @@
 void AyStar::OpenListAdd(PathNode *parent, const AyStarNode *node, int f, int g)
 {
 	/* Add a new Node to the OpenList */
-	PathNode *new_node = this->nodes.CreateNewNode();
-	new_node->Set(parent, node->tile, node->td, true);
-	new_node->estimate = f;
-	new_node->cost = g;
-	this->nodes.InsertOpenNode(*new_node);
+	PathNode& new_node = this->nodes.CreateNewNode();
```
Comment (LordAro) [inline src/pathfinder/aystar.cpp]:
```suggestion
	PathNode &new_node = this->nodes.CreateNewNode();
```
;)

File: src/pathfinder/yapf/nodelist.hpp [hunk 2]
```diff
@@ -15,38 +15,28 @@
...
+template <class Titem, int Thash_bits_open, int Thash_bits_closed>
+class NodeList {
 public:
-	typedef Titem_ Titem;                                        ///< Make #Titem_ visible from outside of class.
-	typedef typename Titem_::Key Key;                            ///< Make Titem_::Key a property of this class.
-	using CItemArray = std::deque<Titem_>;                       ///< Type that we will use as item container.
-	typedef HashTable<Titem_, Thash_bits_open_  > COpenList;     ///< How pointers to open nodes will be stored.
-	typedef HashTable<Titem_, Thash_bits_closed_> CClosedList;   ///< How pointers to closed nodes will be stored.
-	typedef CBinaryHeapT<Titem_> CPriorityQueue;                 ///< How the priority queue will be managed.
+	using Item = Titem;
```
Comment (LordAro) [inline src/pathfinder/yapf/nodelist.hpp:23]: Is that a good idea? the 'T' prefix is there for a reason...

(ok, it's not there for the Key typedef, but I guess that's not actually a template param)

Comment (Kuhnovic) [reply src/pathfinder/yapf/nodelist.hpp:23]: It's not really a template parameter anymore if you use a typedef / alias, and our style guide doesn't specifically describe what to do in this case. IMO the non-prefixed version makes more sense, but this is a bit of a grey area.

**Review by LordAro — APPROVED** [review]
:shipit:

**Pull request merged**